### PR TITLE
Add tests for MHC on classy clusters

### PR DESCRIPTION
**Pull request by tenczar**
### What this PR does / why we need it
This PR adds MachineHealthCheck tests to the CC test suite. The tests exercise the new behavior of the machinehealthcheck API when using classy clusters. In particular, create, update, and delete are not supported for classy clusters and those those operations now return errors. Machine health checks must be modified at the clusterclass level for classy clusters. This change also updates the tanzu cli to properly retrieve machine health checks for nodes in either legacy clusters or classy clusters.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
E2E tests retrieve machine health checks and then verifies errors on set/delete operations. The test also causes a node failure state and waits for the cluster remediation to finish.

### Release note

```release-note
Modifying machinehealthchecks on classy clusters is not supported. The API will return an error when performing these operations on classy clusters.
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/shared/mhc_cc.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		if tkgCtlClient2 == nil {
+			_, _ = GinkgoWriter.Write([]byte("tkgCtlClient is nil"))
+		}
+		err = tkgCtlClient2.DeleteMachineHealthCheck(tkgctl.DeleteMachineHealthCheckOptions{
+			ClusterName:            clusterName,
+			MachinehealthCheckName: mhc.Name,
+			Namespace:              namespace,
+			SkipPrompt:             true,
+		})
+		Expect(err).To(HaveOccurred())
```
Comment (anujc25) [inline pkg/v1/tkg/test/tkgctl/shared/mhc_cc.go:140]: We should also add a check to make sure we are getting the error message here by comparing the output with the error message that we are expecting.
```
Expect(err.Error()).To(ContainSubstring("deleting machine health checks on clusterclass based clusters is not supported"))
```

File: pkg/v1/tkg/test/tkgctl/shared/mhc_cc.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+		// Set MHC and verify if it is set
+		By(fmt.Sprintf("Updating MHC for cluster %q", clusterName))
+		err = tkgCtlClient2.SetMachineHealthCheck(tkgctl.SetMachineHealthCheckOptions{
+			ClusterName:            clusterName,
+			Namespace:              namespace,
+			MachineHealthCheckName: mhc.Name,
+			UnhealthyConditions:    fmt.Sprintf("%s:%s:%s", string(corev1.NodeReady), string(corev1.ConditionFalse), "5m"),
+		})
+		Expect(err).To(HaveOccurred())
```
Comment (anujc25) [inline pkg/v1/tkg/test/tkgctl/shared/mhc_cc.go:150]: Same as above
```
Expect(err.Error()).To(ContainSubstring("setting machine health checks on clusterclass based clusters is not supported"))
```

**Review by anujc25 — APPROVED** [review]
Looks good.

Couple of minor comments to also test the error message as part of e2e test to make sure we are seeing the error that we are expecting.

File: pkg/v1/tkg/test/tkgctl/shared/mhc_cc.go (hunk 1 above)
Comment (tenczar) [reply pkg/v1/tkg/test/tkgctl/shared/mhc_cc.go:140]: done

**Pull request merged**